estate/models/estate_property.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate Property"
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(default=fields.Datetime.add(fields.Datetime.now(),months=3),copy=False)
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True,copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Orientation',
+        selection=[('North', 'North'), ('South', 'South'), ('East', 'East'), ('West', 'West')],
+        help="test")
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        string='State',
+        selection=[('New', 'New'), ('Offer Received', 'Offer Received'), ('Offer Accepted', 'Offer Accepted'),
+                   ('Sold', 'Sold'), ('Canceled', 'Canceled')],
+        default='New',
+        copy=False
+        )
+    property_type_id = fields.Many2one("estate.property.type")
+    salesman_id = fields.Many2one("res.users", string="Salesman", index=True, default=lambda self: self.env.user)
+    buyer_id = fields.Many2one("res.partner", string="Buyer")
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+    offer_ids = fields.One2many("estate.property.offer","property_id")
+    total_area = fields.Float(compute="_compute_total_area")
+    best_price = fields.Float(compute="_compute_best_price")
+
+    @api.depends("living_area", "garden_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids")
+    def _compute_best_price(self):
+        self.best_price = 0
+        for record in self.offer_ids:
+            if self.best_price < record.price :
+                self.best_price = record.price
+
```

Comment:
Not sure if it works properly if we change the price on an offer that already exists. Is it gonna get triggered ?